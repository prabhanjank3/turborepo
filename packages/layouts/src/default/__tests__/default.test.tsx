import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import DefaultLayout, { DefaultLayoutProps } from '..';
import { MemoryRouter } from 'react-router-dom';

// Mock Navbar and Footer components
const MockNavbar = () => <div data-testid="navbar">Mock Navbar</div>;
const MockFooter = () => <div data-testid="footer">Mock Footer</div>;

const renderWithRouter = (ui: React.ReactElement) => {
  return render(<MemoryRouter>{ui}</MemoryRouter>);
};

describe('DefaultLayout Component', () => {
  const defaultProps: DefaultLayoutProps = {
    Navbar: <MockNavbar />,
    Footer: <MockFooter />,
    children: <div data-testid="main-content">Main Content</div>,
  };

  it('renders the Navbar component', () => {
    renderWithRouter(<DefaultLayout {...defaultProps} />);
    expect(screen.getByTestId('navbar')).toBeInTheDocument();
  });

  it('renders the Footer component', () => {
    renderWithRouter(<DefaultLayout {...defaultProps} />);
    expect(screen.getByTestId('footer')).toBeInTheDocument();
  });

  it('renders the main content', () => {
    renderWithRouter(<DefaultLayout {...defaultProps} />);
    expect(screen.getByTestId('main-content')).toBeInTheDocument();
    expect(screen.getByText('Main Content')).toBeInTheDocument();
  });

  it('renders correctly with all components', () => {
    const { container } = renderWithRouter(<DefaultLayout {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
