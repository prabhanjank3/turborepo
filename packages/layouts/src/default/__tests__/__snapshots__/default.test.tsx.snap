// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultLayout Component renders correctly with all components 1`] = `
<div>
  <div
    class="MuiBox-root css-zf0iqh"
  >
    <div
      data-testid="navbar"
    >
      Mock Navbar
    </div>
    <div
      class="main-content MuiBox-root css-i9gxme"
    >
      <div
        data-testid="main-content"
      >
        Main Content
      </div>
    </div>
    <div
      data-testid="footer"
    >
      Mock Footer
    </div>
  </div>
</div>
`;
